### ISSUE TITLE
"ohai requires Ruby version >= 2.4"  with gem installs and "chef-zero requires Ruby version >= 2.4.3" with bundle installs

### ISSUE THREAD

**Issue by lamont-granquist** [berkshelf/berkshelf#1755]
This is due to the release of chef/ohai-14 on 4/3/18 affecting users doing manual gem/bundler installs on ruby-2.3

See:

https://github.com/bundler/bundler/issues/6471

As we bumped those floors in major versions this isn't really a bug that we can fix in code on our side.  I believe those are bugs in rubygems/bundler which should be pruning gem versions that don't install on the current ruby version out of the possible solution set.  The workarounds for users are:

1.  Use straight ChefDK -- this is why it exists as a thing.  You can look at the chef-cookbooks travis configs for examples of how we install chefdk on travis and use it to test our own cookbooks (any .travis.yml in the chef-cookbooks organization should do).  This means you get older versions of tools, but that's the point -- you get tools which are compiled together and all work together as a set.

2.  Use Gemfile and pin the chef version manually:

```
source "https://rubygems.org"
ruby "2.3.7"
gem "berkshelf"
gem "chef", "~> 13"
```

3.  Upgrade to ruby 2.4/2.5 (this is the happiest of all paths)

**Comment by lamont-granquist** [comment]
okay after more coffee, looks like i fixed this by releasing berkshelf 6.3.2 which pins to `chef < 14`

**Comment by lamont-granquist** [comment]
yeah think this is worked around / fixed.
